packages/flutter/test/foundation/diagnostics_test.dart

```diff
@@ -1180,7 +1180,11 @@ void main() {
       ifNull: 'disabled',
     );
 
-    expect(present.toString(), equals('onClick: Closure: () => void'));
+    if (!identical(0, 0.0)) {
+      expect(present.toString(), equals('onClick: Closure: () => void'));
+    } else {
+      expect(present.toString(), equals('onClick:\n  Closure: () => void from: function onClick() {\n        }'));
```

Comment:
that said, the `\n` after the `:` is distressing. Indeed all the `\n`s here are distressing...

Reply:
/cc @vsmenon Can this be fixed on the ddc/dart2js side? I can't think of a good reason for the VM and the web compiler to deviate from each other.

Reply:
@yjbanov - do you mind filing an sdk bug?  

This does look fixable.  (The rtti code is different, so that may not be always true.)

Reply:
Filed: https://github.com/dart-lang/sdk/issues/37105